refactor: rework introduction and related sections (#113)
eclipse-dataspace-protocol-base/DataspaceProtocol: README.md

# Dataspace Protocol

The __Dataspace Protocol__ is a set of specifications designed to facilitate interoperable data sharing between entities governed by usage control and based on Web technologies. These specifications define the schemas and protocols required for entities to publish data, negotiate [Agreements](https://eclipse-dataspace-protocol-base.github.io/DataspaceProtocol/#dfn-agreement), and access data as part of a federation of technical systems termed a [Dataspace](https://eclipse-dataspace-protocol-base.github.io/DataspaceProtocol/#dfn-dataspace).

The web rendering of the spec represents the current state on this repo's main branch:
https://eclipse-dataspace-protocol-base.github.io/DataspaceProtocol/.

## About versions

> Since [Version 0.8](https://github.com/eclipse-dataspace-protocol-base/DataspaceProtocol/tree/main/releases/v0.8), the specification has been stable with changes in details. The latest release candidate ([2024-1](https://docs.internationaldataspaces.org/ids-knowledgebase/dataspace-protocol)) of the Dataspace Protocol specification is considered stable. It has been developed under the governance of the [International Data Spaces Association](https://internationaldataspaces.org/). Further changes shall not affect conformity. 

Reviewer: We have to be very careful here. The specification has been developed under Eclipse governance, not IDSA

Reviewer 2: I'd argue that 2024-1 is not the "latest release candidate" but rather the latest release. I'd rather be careful with claims toward conformity.

Reviewer 3: I think we should leave the version history here and add a sentence clarifying that after 2024-1 the project changed the governance from IDSA to Eclipse.

Author: Always happy about discussions about _moved_ content, especially from its authors.  

Let me come up with a better proposal.

Reviewer 3: Acknowledged, but for me not possible during this week.

Author: I've added a new commit:

> Previous versions of the Dataspace Protocol ([0.8](https://github.com/International-Data-Spaces-Association/ids-specification/releases/tag/v0.8) and [2024-1](https://github.com/International-Data-Spaces-Association/ids-specification/releases/tag/2024-1)) have been developed and released under the governance of the [International Data Spaces Association (IDSA)](https://internationaldataspaces.org/).

> Subsequent versions of the Dataspace Protocol are maintained and developed within the [Eclipse Dataspace Protocol project](https://projects.eclipse.org/projects/technology.dataspace-protocol-base) associated with the [Eclipse Dataspace Working Group](https://dataspace.eclipse.org/), under the governance of the [Eclipse Foundation (EF)](https://www.eclipse.org/). Version [2024-1](https://github.com/International-Data-Spaces-Association/ids-specification/releases/tag/2024-1) marks the initial contribution to the [Eclipse Dataspace Protocol project](https://projects.eclipse.org/projects/technology.dataspace-protocol-base) by the IDSA.

WDYT?

> Now, the spec is maintained and developed by the [Eclipse Dataspace Protocol](https://projects.eclipse.org/projects/technology.dataspace-protocol-base) project under the purview of the [Eclipse Dataspace Working Group](https://dataspace.eclipse.org/) (EDWG). The source of truth is this repository. 

Reviewer: ```suggestion
The spec is maintained and developed by the [Eclipse Dataspace Protocol](https://projects.eclipse.org/projects/technology.dataspace-protocol-base) project under the purview of the [Eclipse Dataspace Working Group](https://dataspace.eclipse.org/) (EDWG). 
```

Reviewer 3: Please change to [Eclipse Dataspace Protocol Project], as it is a project and the result is still the Dataspace Protocol.

> __NOTE:__ 

> > A versioning scheme beside the commits to the repository is not available but will be provided in the future.

Reviewer 2: ```suggestion
```

## Best practices

> The Dataspace Protocol is under development and the working group is active on this draft, reviewed and improved the content multiple times. During the process, several aspects were discussed, which are not considered part of the _normative_ specification, but important to be documented as support for the users of this specification as **best practices**. The [DSP Best Practices](https://github.com/eclipse-dataspace-protocol-base/dsp_best_practices) are _non-normative_.

Reviewer: ```suggestion
Additional, non-normative information related to this specification can be found in the [DSP Best Practices Guide](https://github.com/eclipse-dataspace-protocol-base/dsp_best_practices). 
```
